From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Basil Chew <basil.chew@intel.com>
Date: Mon, 22 Aug 2022 14:44:41 +0800
Subject: [PATCH] Update logic to add GVT-d grub cmdline

Change:
- updated logic to add GVT-d grub cmdline only for non-SRIOV case

Change-Id: Ia9c7b92580086a43863244133f517dc067280985
---
 scripts/setup_host.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
index fb5eb57..c7950b7 100755
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@ -120,7 +120,9 @@ function ubu_build_ovmf_gvt(){
 }
 
 function ubu_enable_host_gvt(){
-    if [[ ! `cat /etc/default/grub` =~ "i915.enable_gvt=1" ]]; then
+
+    if [[ ! `cat /etc/default/grub` =~ "i915.enable_guc=0x7 udmabuf.list_limit=8192" ]] &&
+       [[ ! `cat /etc/default/grub` =~ "i915.enable_gvt=1" ]]; then
         read -p "The grub entry in '/etc/default/grub' will be updated for enabling GVT-g and GVT-d, do you want to continue? [Y/n]" res
         if [ x$res = xn ]; then
             return
-- 
2.17.1

